### PR TITLE
Refactor the DialogService

### PR DESCRIPTION
**Pull request by willsb**
No reason for it not to be awaitable.

Squash whenever.

### CODE REVIEW

File: Toggl.Daneel/Services/DialogService.cs [hunk 1]
```diff
@@ -1,36 +1,42 @@
 ﻿using System;
```
Comment (arturs105) [inline Toggl.Daneel/Services/DialogService.cs]: 🔥

File: Toggl.Daneel/Services/DialogService.cs [hunk 2]
```diff
@@ -1,36 +1,42 @@
...
+            var alert = UIAlertController.Create(title, message, UIAlertControllerStyle.Alert);
+            alert.AddAction(UIAlertAction.Create(dismissButtonText, UIAlertActionStyle.Cancel, _ => tcs.SetResult(null)));
             getPresentationController().PresentViewController(alert, true, null);
+
+            return tcs.Task;
         }
 
-        public void ShowMessage(string title, string message, string dismissButtonTitle, Action dismissAction)
+        public Task<bool> Confirm(
+            string title, string message,
```
Comment (arturs105) [inline Toggl.Daneel/Services/DialogService.cs]: I'd like these to be on separate lines

File: Toggl.Foundation.MvvmCross/Services/IDialogService.cs [hunk 3]
```diff
@@ -1,21 +1,20 @@
 ﻿using System;
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/Services/IDialogService.cs]: 🚫

File: Toggl.Daneel/Services/DialogService.cs [hunk 4]
```diff
@@ -1,36 +1,42 @@
...
     {
-        public void Confirm(
-            string title,
-            string message,
-            string confirmButtonTitle,
-            string dismissButtonTitle,
-            Action confirmAction,
-            Action dismissAction,
-            bool makeConfirmActionBold)
+        public Task Alert(string title, string message, string dismissButtonText)
```
Comment (simonrozsival) [inline Toggl.Daneel/Services/DialogService.cs]: I can't see any dialog like this in Zeplin and you didn't use it anywhere in this PR either. Do we need this at all? Can you think of an use case?

Comment (willsb) [reply Toggl.Daneel/Services/DialogService.cs]: There are use cases, but YAGNI wins :P

**Pull request merged**